In [1]:
%%file arreglo_x.libertad
424329
424448
424652
424665
424257
424453
424425
424530
424466
424566
424495
424603
424518
424644
424539
424660
424556
424667
424558
424677
424559
424687
424683
424645
424622
424660
424663
424476
424589
424625
424657
424296
424540
424559
424447
424423
424568
424642
424505
424716
424640
424943
424938
425043
424979
424864
424866
424658
424827
424615
424472
424650
424766
424630
424715
424563
425014
424659
424862
424726
424704
424802
424897
424796
424696
424843
424639
424913
424828
424922
424697
424604
424843
424749
424626
424623
424786
424621
424808
424650
424781
424916
424889
424825
425076
424869
424926
424921
425051
424884
424970
425170
425177
425066
425099
424851
425177
425290
425175
425279
425256
425194
425265
425182
424989
425187
425184
425543
425183
425087
425187
425166
424996
424994
425310
425248
425272
425441
425380
425474
425499
425627
425079
424904
424844
424653
425165
424853
425060
425052
424993
424906
425011
424743
425253
424986
425087
424950
425076
425187
425514
425311
426103
424162
425167
425063
425064
425069
425001
425080
425046
425165
424811
424882
425131
424868
425501
425620
425587
425856
425626
425910
425401
425808
425729
425880
426068
425940
425718
425996
425704
425285
425693
425999
425880
426096
425835
425863
425777
425980
425754
425961
425937
425991
425982
426027
425856
425927
425805
425844
426131
425822
426251
426309
426185
426246
426132
426189
425994
426051
426059
426251
425612
425623
425589
425621
424239
424278
424407
424331
424368
424407
424332
424369
424408
424364
424346
424385
424148
424248
424319
424239
426386
426448
424140
424292
424290
424193
424260
424159
424140
424237
424112
424170
424144
424212
424112
424152
424136
424181
425321
425724
425826
425581
426250
426577
425747
425776
425636
425721
425683
425822
425479
425758
425568
425644
425634
425656
425620
425646
425690
425772
425700
425762
425709
425779
426258
426410
425721
425791
425732
425800
425806
425743
425816
425755
425634
425705
425391
425667
425580
425414
425563
425582
425510
425524
425358
425422
426132
426786
425343
425323
425340
425370
425737
425894
425897
425720
425721
425595
425594
425629
425686
425624
425618
425697
425672
425722
425720
425693
426164
426007
425646
425697
425499
425584
424817
424571
424837
424811
424974
424834
426149
426573
426419
426522
426467
426371
426627
426690
426913
426850
426923
426996
427032
426957
427010
427528
426776
426943
427438
427482
427473
427533
427156
427545
427019
427344
427371
427453
427298
427570
427229
427273
426118
426223
426083
426027
426063
425945
425982
426141
426075
426055
426070
426103
425544
425476
426121
425850
426475
426106
426505
426364
425407
425492
425371
425435
425691
425384
425678
425933
425922
426115
426092
426434
426100
426390
426431
426430
424335
424344
424335
424500
424501
424560
424558
424561
424639
424950
425214
425305
425217
425262
425255
425318
425549
425665
425599
425700
425744
425448
425730
425834
426157
426079
425902
426091
426023
426106
426103
426159
426120
426099
426070
426116
426153
426159
426208
426347
426343
426484
426482
426524
426118
426729
426510
426590
426375
426531
426400
426588
426538
426625
426540
426758
426586
426894
426652
426982
426492
426655
426520
426495
426329
426538
426591
426900
426723
426826
427016
426903
426764
427047
426834
426929
426601
426743
426746
426835
426601
426703
426687
426683
426733
426768
426766
426705
426675
426732
426700
426898
426892
427010
427001
427093
426450
426939
426930
427049
427033
427077
426517
426623
426627
426963
426626
426890
426793
427088
426888
427044
427034
427200
427184
427231
427215
427376
427160
427307
426943
427164
426919
426958
427011
427104
426918
427315
426912
426984
426982
427045
426965
427283
427178
427372
427184
427183
426754
426751
426586
426696
426763
427077
427075
427193
427188
427247
427242
427371
427590
427396
427408
427240
427285
427486
427488
427711
427326
427623
427615
427683
427619
427632
427551
427583
427117
427167
427118
427049
427247
426974
425869
425690
425549
425604
425509
425688
426037
426042
425534
426123
425298
425552
425534
425569
425584
425623
425583
425746
425588
425457
425372
425450
425376
425535
425530
426009
425445
425621
425593
425696
425642
425548
425520
425557
425478
425706
425317
425980
425319
425439
425501
425329
425329
425377
425320
425333
425321
425337
425278
425235
425236
425295
425275
425414
425263
425337
425276
425265
425774
425832
425246
425768
425203
425644
425639
425796
425682
426128
425517
425860
425855
425860
425861
425916
425404
425491
425438
425379
425379
425421
426158
426248
426363
426244
425794
425858
425797
426006
425315
424823
424108
424707
424687
424925
424894
425332
425327
425540
425509
426021
424237
424833
424050
424264
424060
424488
424001
424081
423952
424083
423895
423955
423842
423966
423942
424362
423955
424063
424062
424257
424482
424686
424682
425096
424400
425227
424311
424701
424694
424785
424782
424947
424541
424878
424763
425078
425069
425338
425074
425257
425249
425345
425304
425522
425440
425512
425437
425680
425494
425554
425395
425579
425400
425437
424514
424365
424431
424514
424358
424442
424337
424363
424241
424347
424558
424336
424553
424553
424552
424506
424505
424841
424823
424976
425299
426224
423797
424348
424331
424639
424203
424341
423804
424015
423942
424030
423791
423839
424145
424415
424156
424191
424178
424238
424210
424246
423928
424155
423930
424032
424039
424026
423945
423838
423841
423792
423864
424181
424170
424208
423997
424219
424047
424162
424113
424159
424155
424202
424251
424296
424222
424472
424243
424362
424526
424712
424415
424943
424519
424631
424476
424652
424378
424754
424343
424380
424352
424474
424442
424813
424591
424724
424664
424778
424291
424405
424400
424487
425012
428859
426138
426620
426142
427273
427218
428239
427206
427443
427006
427223
426538
427108
426627
427165
426310
427035
426550
427063
426364
426079
425975
427620
427308
427405
427375
427502
427325
427444
427430
427477
427473
427584
427014
427106
426598
427708
427003
427819
427027
427772
427032
426603
426627
427004
426309
426510
426613
426219
425974
426231
426751
426834
427742
427853
427547
427896
427878
428688
427891
428986
427662
428023
427751
428114
427444
427766
427533
427612
427603
427644
427638
427771
427638
428108
427955
427984
427980
428126
427907
427988
427604
428095
427670
427757
427674
427800
427756
427887
427684
427879
427647
427776
427624
427752
428614
428533
428519
428604
428291
428541
428433
428429
428300
428277
428099
428435
428429
428708
428693
428713
428586
428626
428284
428783
428444
428442
428368
428359
428210
428202
427482
427597
427577
427944
427862
427939
427887
427975
427893
427955
427802
427823
427848
427953
427898
428123
427986
428050
427849
428019
427893
427936
428007
428022
428023
428070
428065
428153
428119
428200
428075
428117
428156
428177
427918
428163
427924
427835
427836
427979
427615
427764
427751
427845
427833
427992
427710
427620
427622
427822
427693
427805
427631
427652
427679
427706
427728
427789
427468
427751
427672
427752
427685
427787
427783
427851
427763
427796
427792
427892
427792
427823
427922
428211
428206
428300
428375
428484
428286
428475
428303
428503
428494
428501
427965
428270
428200
428287
428134
427927
428140
428205
428201
428354
427517
427543
427614
427637
427636
427577
427470
427512
427465
427515
427446
427505
427444
427551
427543
427626
427574
427535
427549
427537
427478
427589
427617
427583
427618
427786
427980
428089
428084
428144
427845
428033
427991
428012
427877
427640
427579
427655
427580
427584
427285
427824
426625
426965
427424
427657
426910
426920
426854
427098
427296
427088
426980
426992
426751
426989
426800
426809
426832
426879
427041
427135
426970
427021
427038
427409
427039
427071
426938
427000
426918
427015
426826
427023
426950
427011
427007
427145
426071
426647
426193
426128
426131
426191
426251
426188
426491
426634
426126
426210
426052
426406
426185
426287
427257
427492
426579
426750
426604
426633
426634
427014
426593
426662
426520
426583
426450
426507
426507
426622
426708
426776
426660
426830
426648
426770
426656
426525
426733
426881
426905
427018
426736
426844
426798
426973
426758
426926
426762
426681
426647
426689
426739
426765
426764
426847
426844
426957
426957
427094
427052
427177
426739
427178
427090
427242
427017
427256
427112
427308
427229
427384
427321
427407
424984
424246
424940
426482
426459
426690
426686
426766
426757
427012
426989
427141
427132
427311
427319
426952
427136
427129
427291
427407
427302
427463
427154
427316
427416
427296
427189
427369
427352
427476
425050
425290
425073
425312
425096
425331
425028
425132
425089
425166
424962
425065
425278
425420
425294
425393
425249
425475
425163
425175
425172
425239
425169
425217
425167
425200
425160
425254
425250
425281
425268
425328
425326
425533
425279
425452
425435
425834
425826
425927
425832
426128
425339
425452
425369
425468
425434
425516
425481
425513
425455
425513
425404
425468
425530
425864
425617
425929
425508
425569
425574
425639
425646
425686
425662
425768
425764
425792
425703
425809
425796
425873
425792
425834
425791
425838
425791
425872
425760
425810
425761
425783
425780
425826
425788
425809
425048
425147
425125
425420
425361
425448
425433
425517
425505
425674
425682
425739
425254
425359
425063
424876
424718
424764
424744
424729
423904
424074
423950
424014
423951
424076
424010
424160
423944
424067
423930
423906
423944
423943
423921
424074
424223
424314
424185
424286
424282
424302
424302
424282
424095
424225
424072
424403
424199
424718
424935
425155
425119
425458
425863
426125
430207
430590
429920
430548
430365
430765
430743
430621
430850
430747
430928
430825
431004
430904
430825
430975
430994
430377
430406
430390
430400
430328
430343
430417
430546
430465
430669
430528
430534
430420
430337
430431
430383
430191
430322
430258
430295
429970
430043
430090
430182
430088
430264
430139
430333
430196
430146
430098
430203
430158
429709
429908
429771
429716
429889
430257
430036
429988
430086
430037
430028
429961
429933
429985
429950
429904
429980
429973
429777
430054
429926
429827
429874
429976
429868
429766
429934
429926
429918
429723
429410
429416
429477
429480
429548
429476
429501
429408
429476
429686
429673
429762
429871
429705
429376
429796
429767
429704
429729
429636
429649
429582
429631
429554
429486
429561
429338
429626
429619
429733
429570
429659
429651
429673
429673
429586
429273
429422
429296
429423
429208
429305
429132
429383
429027
429352
428862
429039
429302
429415
429418
429315
428655
428746
428697
428794
428742
428862
428662
428788
428694
428802
428863
428788
428719
428877
429038
428868
428695
428960
428598
428647
428505
428598
428799
428578
428594
428464
428473
428234
428255
428372
428158
428313
428332
428437
428570
428670
428611
428517
428522
428439
428447
428435
428435
428307
428495
428522
428292
427937
428051
428152
428019
428120
428096
428183
428250
428199
427938
428060
427701
427977
427779
428056
427877
427982
428131
428039
428034
428089
426079
425976
426200
425999
426465
426711
426282
426498
426469
426470
426823
426907
427490
427367
427268
427504
427235
427278
427370
427387
427792
427862
427795
427813
427266
427153
427159
427042
427046
426867
426878
426761
426769
426409
426857
426874
426948
426870
423930
423964
427605
427783
427709
428536
426889
427639
427667
427786
427781
428238
428074
428226
427952
428490
428347
428216
428222
428416
424294
424634
424448
424620
424622
424686
424638
424922
424835
424902
424862
424992
424915
424843
425165
425254
425538
425230
425255
425005
425108
425308
425470
425275
424995
425092
425087
425420
425007
425069
425006
425273
425127
425082
425082
425119
425180
425110
424649
424591
424593
424632
424594
424614
424609
424779
424764
424779
424577
424781
424702
424954
424937
425047
424680
425053
424884
424975
424972
424886
424888
425050
425091
425114
424621
424795
424783
424624
426997
426815
426824
426915
426119
426122
425686
426122
426164
426136
426119
426136
426125
426122
426115
426047
425848
426094
426057
426004
426046
426062
425449
425883
425437
425456
426409
426838
426119
426235
426094
426472
426125
426103
426066
426419
426208
426426
426101
426217
426220
426258
426251
426305
426316
426338
426333
426356
426361
426377
426360
426399
426393
426406
426201
426535
426101
426072
426094
426196
426208
426567
426404
426759
426338
426523
427690
427708
428114
428166
428137
428357
427993
428095
428025
428127
428048
428004
428067
428018
428070
428130
428054
428117
428043
428109
428087
428151
428152
428096
428091
428072
428141
428105
428185
428154
428138
428175
428119
428160
428118
428326
428147
428335
428414
428322
428421
428331
426086
425852
425882
425631
425643
425362
424189
424302
424541
424292
425376
425200
425300
425149
424937
425051
425232
424736
424746
424607
424620
424365
424383
424330
424403
424402
424279
424383
424320
424541
424479
424603
424883
425210
425110
425145
424843
425232
424858
424625
424814
424951
424895
424977
424713
424750
424745
424841
424751
424757
424711
424817
424907
424939
424861
424893
424874
424910
424887
425007
424711
424992
424726
424799
424861
424818
424816
424811
424810
424769
424811
424907
424774
424903
424816
425004
424638
424964
424572
424652
424653
424700
424687
424790
424683
424492
424567
424637
424583
424718
424512
424578
424637
424659
424579
424632
424613
424679
424227
424462
424433
424646
424591
424634
424384
424461
424436
424467
424482
424528
424488
424525
424476
424511
424614
424734
424630
424786
424771
424952
424750
424653
424942
425130
424974
425156
425127
425319
425143
425326
425323
425573
425310
425545
425539
425647
425553
425667
425635
425749
425665
425796
426042
425998
425799
425971
425983
425924
425788
425887
425888
426000
425919
426004
426002
425987
425918
425966
425669
425721
425425
425527
425520
425632
425544
425599
425537
425728
425626
425674
425539
425727
425720
425769
425622
425825
425835
425958
425848
425950
425853
425942
425840
425940
425750
425653
425678
425745
425699
425761
425717
425777
425733
425800
425756
425821
425755
425932
425827
425907
425847
425924
425859
425938
425876
425953
425891
425962
425906
425959
425819
425881
425789
425836
425806
425853
425834
425899
425850
425932
425864
425948
425881
425941
425676
425741
425682
425746
425678
425684
425593
425739
425606
425710
425503
425533
425539
425579
425642
425572
425561
425606
425680
425722
425667
425706
425703
425748
425687
425735
425722
425823
425446
425502
425497
425580
425505
425564
425286
425355
425277
425338
425264
425328
425254
425319
425161
425250
425214
425308
425108
425177
425141
425196
425246
425294
425262
425326
425379
425389
425346
425233
425247
425301
425364
425457
425386
425478
425476
425559
425491
425548
425491
425421
425672
425755
425735
425748
425544
425336
425389
425469
425434
425430
425482
425494
425460
425467
425463
425497
425500
425565
425538
425547
425546
425625
425681
425564
425650
425686
425635
425692

Writing arreglo_x.libertad


In [2]:
with open("arreglo_x.libertad") as f:
  arr_x = []#arreglo
  for line in f:
    if line == "-\n":
      arr_x.append([])
    else:
      arr_x.append([int(x) for x in line.split()])
arr_x2=[]
for i in arr_x:
  arr_x2.append(i[0])
arr_x=arr_x2
#print(arr_x)

In [3]:
%%file arreglo_y.libertad
4580567
4580577
4580667
4580555
4580489
4580509
4580318
4580385
4580263
4580331
4580217
4580274
4580161
4580191
4580102
4580122
4580057
4580079
4580022
4580028
4579984
4579992
4580001
4579768
4579991
4579840
4579858
4579763
4579989
4579835
4580659
4580641
4579887
4579801
4580501
4580869
4580509
4580362
4580408
4580525
4580554
4580094
4580318
4580236
4580370
4580202
4580220
4580092
4580399
4580239
4580462
4580547
4580505
4580244
4580403
4580324
4580129
4580067
4580055
4579975
4580080
4579924
4579987
4579925
4580230
4579947
4580167
4580434
4580257
4580428
4580930
4580856
4580718
4580920
4580877
4580836
4580664
4580844
4580803
4580660
4580840
4580914
4580815
4580867
4580757
4580777
4580775
4580687
4580804
4580823
4580903
4580606
4580888
4580759
4580722
4580587
4580733
4580720
4580691
4580676
4580683
4580590
4580879
4580778
4580906
4580778
4580782
4580759
4580785
4580723
4580783
4580605
4580584
4580771
4580868
4580566
4580597
4580482
4580675
4580506
4580685
4580539
4580287
4580415
4580729
4580550
4580618
4580720
4580658
4580570
4580576
4580408
4580139
4580505
4580558
4580161
4580488
4580478
4580544
4580449
4580687
4580491
4579257
4578234
4580415
4580484
4580493
4580387
4580574
4580482
4580513
4580616
4580404
4580524
4580360
4580541
4580340
4580458
4580268
4580514
4580240
4580446
4580174
4580567
4580509
4580340
4580233
4580065
4580166
4580327
4580728
4580346
4580475
4580137
4580447
4580516
4580594
4580525
4580567
4580639
4580608
4580684
4580725
4580573
4580722
4580586
4580733
4580547
4580726
4580633
4580631
4580516
4580671
4580588
4580628
4580543
4580596
4580507
4580839
4580726
4580211
4580334
4578611
4578574
4578586
4578593
4577933
4577887
4577822
4577775
4577801
4577716
4577782
4577698
4577724
4577703
4577790
4577706
4577872
4577764
4577821
4577767
4580238
4580137
4577863
4577615
4577627
4577580
4577677
4577624
4577665
4577717
4577783
4577826
4577720
4577757
4577788
4577718
4577805
4577734
4577550
4577149
4577532
4577275
4580837
4580110
4577454
4577425
4577468
4577556
4577414
4577552
4577386
4577639
4577300
4577378
4577373
4577348
4577358
4577333
4577583
4577494
4577569
4577633
4577558
4577627
4580571
4580372
4577544
4577617
4577533
4577601
4577585
4577520
4577568
4577508
4577474
4577400
4577471
4577719
4577662
4577511
4577655
4577636
4577601
4577585
4577581
4577513
4580143
4580709
4577590
4577544
4577578
4577572
4577706
4577504
4577513
4577148
4577155
4577028
4577031
4576991
4577048
4576992
4577055
4576972
4577041
4576989
4576996
4576970
4580356
4580274
4577085
4577030
4577408
4577312
4577071
4577167
4577102
4577071
4577308
4577091
4580353
4580577
4580572
4580647
4580601
4580752
4580449
4580519
4580492
4580414
4580348
4580423
4580383
4580314
4580379
4580652
4580697
4580807
4580780
4580656
4580793
4580588
4580377
4580607
4580807
4580266
4580800
4580439
4580705
4580754
4580801
4580672
4580842
4580637
4580464
4580333
4580401
4580357
4580313
4580399
4580361
4580403
4580439
4580377
4580223
4580142
4580575
4580484
4580807
4580573
4580763
4580677
4579983
4579833
4580097
4580133
4580204
4580055
4580201
4580408
4580395
4580471
4580450
4580679
4580559
4580104
4580833
4580771
4580652
4580467
4580474
4580218
4580223
4580055
4580059
4579982
4579773
4579909
4580406
4580239
4580398
4580453
4580439
4580505
4580702
4580574
4580720
4580610
4580661
4580378
4580644
4580682
4580783
4580752
4580710
4580733
4580736
4580475
4580468
4580350
4580631
4580450
4580803
4580621
4580359
4580271
4580535
4580319
4580330
4580167
4580174
4580093
4580036
4580471
4580481
4580366
4580117
4580098
4580108
4580253
4580339
4580130
4580340
4580084
4580385
4580042
4580423
4580048
4580670
4580415
4580741
4580659
4580774
4580331
4580269
4580514
4580454
4580559
4580621
4580554
4580594
4580785
4580648
4580616
4580804
4580667
4580762
4580631
4580793
4580789
4580822
4580717
4580670
4580590
4580607
4580536
4580623
4580557
4580546
4580383
4580394
4580271
4580275
4580177
4580818
4580196
4580211
4580135
4580141
4580093
4580102
4580145
4580146
4580450
4580150
4580052
4580080
4580347
4580047
4580141
4580138
4580268
4580259
4580323
4580320
4580429
4580384
4580245
4580738
4580374
4580801
4580720
4580621
4580676
4580465
4580703
4580617
4580498
4580506
4580462
4580421
4580603
4580764
4580423
4580803
4580750
4580819
4580748
4580640
4580719
4580607
4580334
4580339
4580259
4580269
4580194
4580212
4579995
4580653
4580375
4580404
4580197
4580792
4580070
4580084
4579600
4580636
4580695
4580691
4580787
4580787
4580711
4580791
4580636
4580308
4580381
4580313
4580221
4580209
4580053
4580719
4580697
4580697
4580719
4580754
4580709
4580140
4580042
4580133
4580135
4580214
4580134
4579859
4579790
4579881
4579814
4579872
4579973
4579878
4579820
4579860
4579913
4579866
4579782
4579783
4579972
4579914
4579998
4580042
4579846
4580122
4580035
4580126
4580042
4580037
4580038
4580017
4580021
4580211
4579980
4580021
4579929
4579933
4579861
4580019
4579929
4580016
4580065
4580046
4580096
4580093
4580141
4580046
4580180
4580191
4580060
4580211
4580186
4580137
4579900
4580556
4580132
4580486
4580119
4580120
4579886
4580050
4580049
4580564
4580132
4580136
4580049
4580052
4579933
4580175
4580017
4580296
4580236
4580240
4580190
4580265
4580079
4580151
4580080
4580843
4580711
4580309
4579967
4577699
4577533
4577836
4578233
4578202
4578368
4578370
4578616
4578597
4578754
4578750
4579016
4578531
4577531
4578918
4578498
4578468
4577654
4578330
4578434
4578735
4578441
4578867
4578711
4578688
4578723
4578723
4578915
4578443
4578492
4578453
4578526
4577656
4577786
4577778
4578103
4577812
4578437
4579013
4578205
4578232
4578095
4578105
4577982
4578520
4578701
4578644
4578086
4578092
4578347
4578103
4577871
4577886
4577572
4577697
4577753
4577991
4577747
4577987
4578063
4578211
4578018
4578169
4578235
4578174
4578057
4577465
4577277
4577494
4577457
4577453
4577494
4577546
4577451
4577489
4577536
4577620
4577535
4577704
4577615
4577619
4577458
4577457
4577093
4577535
4577291
4578329
4579216
4578833
4579007
4579008
4579067
4579326
4579002
4578950
4579007
4579221
4578902
4579010
4578841
4579035
4579109
4579043
4578954
4578980
4579001
4579057
4578971
4579332
4579035
4579324
4579362
4579437
4579356
4579221
4579187
4579192
4579002
4579509
4579370
4579381
4579295
4579233
4579303
4579170
4579211
4579273
4579205
4578950
4578839
4578979
4578882
4578571
4578690
4578864
4578634
4579050
4578609
4578788
4578991
4578933
4578698
4578935
4578575
4578867
4579014
4578940
4578955
4579015
4578744
4578735
4578877
4579064
4578744
4579073
4578769
4579289
4578969
4578978
4578908
4576993
4580739
4578104
4577451
4577399
4578148
4578146
4578696
4577991
4577576
4577844
4577985
4578490
4577729
4577875
4578273
4578272
4577400
4577974
4578375
4577809
4577535
4577725
4579197
4579237
4579000
4579065
4579088
4579191
4579315
4579354
4579138
4579140
4579162
4578950
4578733
4577474
4578084
4577431
4577799
4577403
4577735
4577408
4577469
4577494
4577443
4577862
4578044
4577460
4577316
4577745
4577311
4578694
4578483
4577718
4577862
4578328
4577792
4577806
4578164
4577793
4578281
4578395
4577848
4578449
4577885
4579048
4578435
4578868
4578908
4578909
4578834
4578838
4578906
4578672
4578923
4578846
4578789
4578791
4578868
4578821
4578664
4579201
4578269
4579187
4579027
4579060
4579126
4578907
4578976
4579352
4578968
4579113
4579179
4579157
4579226
4580705
4580684
4580755
4580486
4580708
4580698
4580708
4580536
4580763
4580167
4580611
4580601
4580603
4580683
4580748
4580627
4580753
4580654
4580496
4580659
4580758
4580698
4580758
4580518
4580769
4580502
4580072
4580135
4580136
4580214
4580563
4580589
4580688
4580476
4580776
4580603
4580612
4580523
4580501
4580541
4580450
4580547
4580446
4580293
4580319
4580387
4580344
4580199
4580772
4580628
4580638
4580521
4580525
4580378
4580547
4580415
4580773
4580542
4580767
4580606
4580697
4580747
4580696
4580650
4580658
4580235
4580307
4580339
4580342
4580384
4580375
4580443
4580787
4580642
4580651
4580185
4580219
4580236
4580313
4580150
4580323
4580162
4580594
4580619
4580403
4580541
4580736
4580546
4580701
4580630
4580600
4580622
4580783
4580522
4580528
4580201
4580524
4580531
4580600
4580702
4580698
4580694
4580530
4580371
4580436
4580225
4580415
4580555
4580567
4580552
4580016
4580369
4580423
4580349
4580311
4580070
4580508
4580538
4580523
4580518
4580555
4580437
4580481
4580431
4580443
4580316
4580407
4580218
4580209
4580224
4580303
4580082
4580300
4580328
4580324
4580316
4580193
4580378
4580447
4580361
4580170
4580201
4580310
4580197
4580306
4580613
4579992
4580099
4580098
4580040
4580197
4579931
4580049
4580020
4579780
4579964
4580142
4579958
4580200
4580136
4579636
4579827
4579189
4579368
4579797
4579968
4579973
4579906
4579957
4579992
4580129
4579989
4579980
4579869
4579883
4579919
4579937
4579890
4579863
4579686
4579891
4579607
4579881
4579737
4579678
4579894
4579683
4579593
4579613
4579641
4579875
4579585
4579672
4579744
4579604
4579410
4579415
4579064
4578986
4579207
4579036
4578247
4578325
4578321
4579059
4578318
4578859
4578567
4578252
4578153
4578729
4578864
4579468
4579249
4579935
4580083
4579902
4579947
4579912
4579792
4579805
4578937
4579796
4579605
4579622
4579472
4579941
4579756
4579770
4579508
4579632
4579655
4579681
4579289
4579534
4579260
4579531
4579441
4579645
4579320
4579340
4579071
4579252
4578991
4579285
4578898
4579681
4579345
4579664
4579822
4579953
4579813
4579955
4579833
4579846
4579631
4579644
4579361
4579373
4579018
4579449
4579139
4578961
4579141
4579482
4579161
4579212
4579307
4579498
4579225
4579594
4579300
4579414
4579459
4578888
4578502
4578879
4579589
4579589
4579687
4579683
4579671
4579670
4579577
4579584
4579612
4579608
4579654
4579642
4579368
4579615
4579497
4579663
4579453
4579667
4579373
4579909
4579631
4579852
4579652
4580064
4579768
4580003
4579697
4579100
4579182
4579061
4579137
4579020
4579095
4579155
4578963
4579176
4578977
4579135
4578932
4579168
4579241
4579278
4579082
4579548
4579120
4579788
4579724
4579727
4579740
4579745
4579752
4579761
4579766
4579787
4579719
4579726
4579537
4579644
4579549
4579560
4579148
4579624
4579626
4579703
4579856
4579860
4579536
4579833
4579974
4579492
4579545
4579569
4579371
4579575
4579393
4579389
4579326
4579302
4579333
4579250
4579285
4579413
4578867
4579459
4578931
4579589
4579418
4579612
4579453
4579642
4579465
4579483
4579303
4579305
4579319
4579491
4579276
4579300
4579336
4579407
4579316
4579403
4579425
4579511
4579334
4579458
4579480
4579462
4579417
4579418
4579441
4579473
4579430
4578933
4578746
4578967
4578442
4578628
4578466
4578680
4578532
4579148
4578829
4578846
4578745
4578575
4578382
4578451
4578841
4578752
4578635
4578696
4578616
4579725
4579538
4579585
4579497
4579577
4579672
4579497
4579611
4579678
4579662
4579828
4579721
4579804
4579681
4579815
4579666
4579594
4579519
4579572
4579617
4579619
4579564
4579570
4579544
4579562
4579531
4579671
4579384
4579306
4579546
4579489
4579504
4579511
4579578
4579721
4579798
4580227
4580709
4580685
4580367
4580586
4580428
4580695
4580388
4580667
4580416
4580668
4580414
4580652
4580388
4580616
4580555
4580426
4580678
4580705
4580669
4580672
4580474
4580525
4580489
4580615
4580405
4580703
4580352
4580358
4580334
4580392
4580334
4580682
4580290
4580487
4580340
4580330
4580428
4580334
4580501
4580695
4580481
4580706
4580467
4580709
4580441
4580474
4580305
4580450
4580297
4580639
4580585
4580695
4580633
4580581
4580419
4580521
4580343
4580685
4580510
4580689
4580452
4580462
4580449
4580460
4580441
4580453
4580424
4580411
4580334
4580692
4580348
4580523
4580494
4580693
4580337
4580371
4580341
4580443
4580497
4580700
4580402
4580589
4580404
4580705
4580580
4580617
4580655
4580578
4580582
4580585
4580531
4580692
4580457
4580411
4580420
4580422
4580355
4580558
4580414
4580587
4580411
4580703
4580477
4580411
4580488
4580501
4580506
4580511
4580492
4580320
4580422
4580415
4580389
4580393
4580307
4580692
4580313
4580275
4580338
4580471
4580272
4580301
4580436
4580355
4580519
4580728
4580356
4580609
4580699
4580578
4580575
4580369
4580210
4580409
4580229
4580211
4580290
4580233
4580299
4580296
4580348
4580423
4580347
4580423
4580266
4580367
4580271
4580585
4580316
4580492
4580340
4580398
4580305
4580493
4580293
4580314
4580231
4580231
4580000
4580156
4580036
4580061
4579955
4580231
4580117
4580302
4580234
4580280
4580224
4580233
4580146
4580158
4580116
4580123
4579958
4580255
4580224
4580493
4580208
4580305
4580229
4580278
4580193
4580339
4580264
4580455
4580509
4580210
4580068
4579006
4579149
4579488
4579761
4579796
4579682
4580038
4579739
4579941
4579883
4579002
4577733
4578173
4578074
4579143
4578639
4579613
4579147
4579218
4579138
4578769
4578551
4578976
4578912
4578825
4578946
4578858
4578823
4578918
4578880
4579305
4579345
4579309
4579275
4579944
4579900
4579904
4579880
4579884
4579859
4579864
4579837
4579842
4579734
4579964
4579856
4579595
4579865
4579638
4579662
4579190
4579507
4579632
4578075
4578558
4577669
4578135
4578216
4578206
4578340
4578610
4578324
4577952
4578185
4578444
4578337
4578338
4578150
4580034
4579504
4580509
4580237
4580249
4579991
4580172
4580435
4580727
4580412
4580590
4580583
4580907
4580683
4580621
4580555
4580763
4580569
4580560
4580584
4580360
4580216
4580387
4580232
4580193
4579587
4579611
4578893
4579769
4579777
4580106
4580183
4580354
4580283
4580291
4580250
4580313
4580252
4579773
4579755
4579829
4579765
4579760
4579678
4579684
4579708
4579713
4579639
4579589
4579648
4579543
4579478
4579897
4579916
4580364
4579703
4579891
4579759
4579775
4579717
4579722
4579714
4580137
4580036
4580858
4580833
4580862
4580718
4580664
4580558
4580550
4580558
4580164
4579965
4580716
4580146
4580285
4580143
4580115
4580147
4579978
4579792
4579807
4579464
4579541
4579397
4579490
4579436
4579476
4579410
4579558
4579734
4579712
4579556
4579747
4578751
4579659
4579508
4579450
4579625
4579668
4579448
4579624
4579744
4579531
4579734
4579488
4579543
4579683
4579568
4579698
4579617
4579607
4579560
4579628
4579568
4579610
4579575
4579605
4579621
4579624
4579591
4579180
4579476
4579377
4579068
4579359
4579184
4579221
4578498
4578813
4578959
4578951
4579040
4577746
4577701
4578028
4577928
4578231
4578012
4578065
4578189
4578115
4578026
4578098
4578046
4578085
4578028
4578169
4578115
4578146
4578097
4578129
4578074
4578187
4578130
4578149
4578080
4578095
4578068
4578211
4578166
4578189
4578152
4578210
4578173
4578187
4578153
4578030
4578245
4578034
4578235
4578312
4578237
4578301
4578228
4579620
4579530
4579530
4579463
4579459
4579325
4580221
4580006
4580130
4580016
4579324
4579196
4579271
4579295
4579499
4579148
4579316
4579155
4579161
4579133
4579141
4579068
4579059
4579017
4579171
4579075
4579138
4579173
4579200
4579331
4579245
4579311
4579155
4579254
4579281
4579171
4579084
4579211
4579091
4579061
4579185
4578862
4579114
4578878
4579278
4579114
4579118
4579132
4579109
4579076
4579313
4579176
4579425
4579324
4579412
4579308
4579320
4579205
4579261
4579298
4579261
4579350
4579083
4578866
4579084
4578937
4578947
4578872
4578880
4578787
4578882
4578870
4578802
4578873
4578805
4578428
4579333
4579437
4579485
4579334
4579342
4579295
4579300
4579334
4579318
4579192
4579370
4579235
4579217
4579276
4579364
4579240
4579249
4579114
4579257
4579058
4579004
4579020
4579259
4579339
4579348
4579372
4579319
4579331
4579395
4579335
4579351
4579284
4579236
4579146
4579149
4579168
4579173
4579191
4579398
4579478
4579372
4579463
4579457
4579467
4579534
4579469
4578891
4578493
4578904
4578508
4578510
4578313
4578517
4578326
4578340
4578226
4578327
4578213
4578227
4578046
4578231
4578053
4578057
4577700
4578065
4577755
4579398
4579002
4578822
4578317
4578655
4577870
4578539
4578572
4578900
4578704
4578930
4578827
4578843
4578721
4578839
4578866
4578694
4578542
4578908
4578737
4578764
4578643
4578576
4578399
4578387
4578452
4578422
4578279
4578239
4578305
4578310
4578157
4578116
4578186
4578192
4578233
4578085
4578114
4578034
4578061
4577994
4578024
4578103
4578071
4578026
4578048
4577981
4578000
4577934
4577954
4577901
4577922
4578171
4577973
4577842
4577932
4578502
4578527
4578457
4578480
4578410
4578434
4578362
4578385
4578316
4578336
4578271
4578290
4578762
4578783
4578714
4578727
4578667
4578682
4578715
4578736
4578668
4578696
4578623
4578651
4578579
4578600
4578243
4578264
4578226
4578244
4578245
4578223
4578357
4578404
4578308
4578342
4578253
4578315
4578458
4578468
4578498
4578474
4578510
4578524
4578539
4578554
4578586
4578598
4578588
4578604
4578634
4578651
4578738
4578768
4577768
4577786
4577789
4577816
4577762
4577779
4577751
4577774
4577783
4577802
4577828
4577850
4577872
4577894
4577988
4578015
4577922
4577954
4578058
4578080
4578016
4578034
4578266
4578116
4578213
4578231
4578302
4578231
4578186
4578151
4578106
4578123
4577920
4577948
4577861
4577892
4577858
4577885
4577806
4577825
4577836
4577814
4578680
4578708
4578744
4578703
4578542
4578333
4578390
4578416
4578355
4578289
4578485
4578452
4578460
4578441
4578444
4578457
4578502
4578521
4578540
4578512
4578557
4578581
4578673
4578558
4578647
4578658
4578630
4578644

Writing arreglo_y.libertad


In [4]:
with open("arreglo_y.libertad") as f:
  arr_y = []#arreglo
  for line in f:
    if line == "-\n":
      arr_y.append([])
    else:
      arr_y.append([int(x) for x in line.split()])

arr_y2=[]
for i in arr_y:
  arr_y2.append(i[0])
arr_y=arr_y2
#print(arr_y)

In [5]:
def convertir(arr_x,arr_y):
    arreglo_f=[]
    i=0
    limite=(len(arr_x))
    while i<limite-1:
        p=[]
        e=i
        limitee=e+2
        while e<limitee:
            p.append(arr_x[i])
            p.append(arr_y[i])
            e+=1
            i+=1
        arreglo_f.append(p)
    return arreglo_f

In [6]:
import math 

def calcular_distancia(x1,y1,x2,y2):
    dist = math.sqrt((x2 - x1)**2 + (y2 - y1)**2 )
    return round(dist,2)

In [7]:
def calcular_pendiente(x1,y1,x2,y2):
    if x2 - x1 != 0:
        m = (y2 - y1) / (x2 - x1)
        es_vertical = False
    else:
        m = None
        es_vertical = True
    return es_vertical, m

In [8]:
def calcular_interseccion(i1,f1,i2,f2):
    f1, m1 = calcular_pendiente(i1[0],i1[1],f1[0],f1[1])
    f2, m2 = calcular_pendiente(i2[0],i2[1],f2[0],f2[1])

    if(m1 is not None and m2 is not None):
        coord_x=(i2[1]-i1[1]-(m2*i2[0])+(m1*i1[0]))/(m1-m2)
        coord_y=(i2[0]-i1[0]-(i2[1]/m2)+(i1[1]/m1))/((1/m1)-(1/m2))
    elif f1 and ~f2:
        coord_x=i1[0]
        coord_y=m2*(i1[0]-i2[0])+i2[1]
    elif f2 and ~f1:
        coord_x=i2[0]
        coord_y=m1*(i2[0]-i1[0])+i1[1]
    return round(coord_x,2),round(coord_y,2)

In [9]:
def ccw(A,B,C):
    return (C[1]-A[1]) * (B[0]-A[0]) > (B[1]-A[1]) * (C[0]-A[0])

# Return true if line segments AB and CD intersect
def is_intersect(A,B,C,D):
    return ccw(A,C,D) != ccw(B,C,D) and ccw(A,B,C) != ccw(A,B,D)

In [10]:
def crear_lista(arreglo,x,y):
  lista_adyiaciencia=[-1]*(len(arreglo))
  for i in range(len(arreglo)):
    r=[]
    for e in range(len(arreglo)):
      continuar=False
      if e==i:
        continue
      if is_intersect([arreglo[i][0],arreglo[i][1]],[arreglo[i][2],arreglo[i][3]],[arreglo[e][0],arreglo[e][1]],[arreglo[e][2],arreglo[e][3]]):
        #r.append(e)
        a,b=calcular_interseccion([arreglo[i][0],arreglo[i][1]],[arreglo[i][2],arreglo[i][3]],[arreglo[e][0],arreglo[e][1]],[arreglo[e][2],arreglo[e][3]])
        x.append(a)
        y.append(b)
        r.append(calcular_distancia(arreglo[i][0],arreglo[i][1],a,b))
    lista_adyiaciencia[i] = r
  return lista_adyiaciencia

In [11]:
def recorrer_inversa(arreglo,recorrido):
  #recorrer_inversa(pesos,recorrido)
  tamanio=len(recorrido)
  a=random.randint(10,50)
  b=random.randint(10,50)
  # a=random.randint(0,10)
  # b=random.randint(0,10)
  if a<b:
    y1=a
    y2=b
  else:
    y2=a
    y1=b
  x1=0
  x2=tamanio
  denominador=x2-x1
  m=(y2-y1)/denominador
  b=y1
  resultados=[-20]*tamanio
  for i in range(tamanio):
    resultados[i]=(i*m)+b
    if resultados[i]==-1:
      print("menos")
  for i in range(tamanio):
    arreglo[recorrido[i][0]][recorrido[i][1]]=int(resultados[i])
  #i+=1
  
  # if i<50:
  #   print_pesos(arreglo)

In [12]:
def asignar_pesos2(arreglo):
  tamanio=len(arreglo)
  arr=[False]*tamanio
  arr_arr=[False]*tamanio
  pesos=[-1]*tamanio
  for i in range(tamanio):
    arr_arr[i]=[False]*len(arreglo[i])
    pesos[i]=[-1]*len(arreglo[i])
  contador=[0]
  recorrido=[]
  
  i=0
  while False in arr:
    #print(arr)
    if i>tamanio:
      i=0
    _dfs2(i,arreglo,arr_arr,arr,pesos,contador,recorrido)
    i+=1
  #print(f'Final arr: {arr}')
  #_dfs2(1,arreglo,arr_arr,arr,pesos,contador,recorrido)
  return pesos

In [13]:
def _dfs2(u,arreglo,arr_arr,arr,pesos,contador,recorrido):
  marcador=True
  for i in arr_arr[u]:
    if not i:
      marcador=False
      break
    # if not arr_arr[u][i]:
    #   marcador=False
    #   break
  if marcador:
    arr[u]=True
  ############
  #print("arr_arr INICIO")
  # for i in arr_arr:
  #   print(i)
  # print("arr_arr FIN")
  ############
  #if not arr[u]:
    ##aquiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiii
  for v in range(len(arreglo[u])):
    if arr_arr[u][v]==False:
        arr_arr[u][v] = True
        recorrido.append([u,v])
        #if v= in arre
        ##########
        for k in range(len(arreglo[arreglo[u][v]])):#estas aquiii
          if arreglo[arreglo[u][v]][k]==u and arr_arr[arreglo[u][v]][k]==False:
            arr_arr[arreglo[u][v]][k]=True#
            #print('aa')
            recorrido.append([int(arreglo[u][v]),k])##
        ###########
        if contador[0]==25:
          contador[0]=0
          recorrer_inversa(pesos,recorrido)
          recorrido=[]
          #¿como recorro de adelante a atrás?
          #¿arreglo de padres y puntos en especifico?
        # else:
        #   contador[0]+=1
        ## hasta aquiiiiiiiiiiiiiiiiiiiiiiiiiiiii
        contador[0]+=1
        _dfs2(arreglo[u][v],arreglo,arr_arr,arr,pesos,contador,recorrido)

In [14]:
def print_pesos(arreglo):
    for i in range(len(aux)):
        print(f'calle {i}: {aux[i]}')

In [15]:
def buscar_vacios(arreglo):
    contador=0
    for i in range(len(arreglo)):
        for e in range(len(arreglo[i])):
            if -1==arreglo[i][e]:
                contador+=1
    print(f'existen {contador} vacios')

In [16]:
def corregir2(arreglo,pesos):
    tamaño=len(arreglo)
    for i in range(tamaño):
        for e in range(len(arreglo[i])):
            if not pesos[i][e]==-1:
                for j in range(len(arreglo[arreglo[i][e]])):
                    if arreglo[arreglo[i][e]][j]==i and not arreglo[arreglo[i][e]][j]==arreglo[i][e]:
                        pesos[arreglo[i][e]][j]=pesos[i][e]

In [17]:
def corregir(pesos):
    for i in range(len(pesos)):
        suma=0
        cont=0
        resultado=0
        if -1 in pesos[i]:
            for e in range(len(pesos[i])):
                if not pesos[i][e]==-1:
                    suma+=(pesos[i][e])
                    cont+=1
            if resultado<0 or cont==0:
                resultado=1
            else:
                resultado=(suma/cont)
            for k in range(len(pesos[i])):
                if pesos[i][k]==-1:
                    pesos[i][k]=int(resultado)

In [18]:
a=10/5
print(a)

2.0


In [19]:
lista=convertir(arr_x,arr_y)
x=[]
y=[]
lista=crear_lista(lista,x,y)
##Lista de distancia entre calle e interseccion
for i in range(len(lista)):
      print(f"Calle {i}: {lista[i]}")
aux=asignar_pesos2(lista)
print(f'lista: {len(lista)} aux: {len(aux)}')
#######
corregir(aux)###################################################aux=pesos lista=arreglo
#######
buscar_vacios(aux)
print_pesos(aux)

Calle 0: [113.48, 10.13]
Calle 1: [8.25, 4.73, 107.08]
Calle 2: [190.5, 85.95, 192.29]
Calle 3: [8.86, 121.52]
Calle 4: [4.2, 116.32]
Calle 5: [4.76, 7.73, 116.69]
Calle 6: [4.53, 122.13]
Calle 7: [4.31, 116.32]
Calle 8: [3.14, 2.1, 110.41]
Calle 9: [1.44, 118.97]
Calle 10: [122.72, 63.88, 30.78, 1.92, 127.02]
Calle 11: [9.47, 155.17, 146.75, 227.76, 232.07]
Calle 12: [3.11, 147.61, 140.64]
Calle 13: [4.06, 7.49, 43.49, 117.58, 75.09]
Calle 14: [3.16, 154.76]
Calle 15: [4.05, 219.86, 321.96]
Calle 16: [84.23]
Calle 17: [75.66, 7.36, 147.36]
Calle 18: [57.82, 1.04, 161.87]
Calle 19: [101.77, 193.61, 5.81]
Calle 20: [62.4, 414.5, 403.09, 236.86, 10.73, 153.8, 526.55, 281.53, 348.07, 282.74]
Calle 21: [4.47, 127.77]
Calle 22: [196.5, 66.35, 7.96, 117.31]
Calle 23: [5.01, 146.64, 241.23, 91.28]
Calle 24: [71.05, 240.01, 5.15, 262.43, 255.59, 146.52]
Calle 25: [106.91, 192.71, 4.87]
Calle 26: [89.58, 282.56, 116.74, 20.69, 172.01]
Calle 27: [83.65, 3.32, 169.62]
Calle 28: [85.63, 311.74, 24

TypeError: list indices must be integers or slices, not float

## Lo que se quiere:
existen 0 vacios-> 1

## lo que se tiene:
existen 170 vacios->5

existen 365 vacios->25

existen 326 vacios->15

In [ ]:
a=[1,2,3,4,5]
print(len(a))
a=[]
print(len(a))

5
0
